lib/index.d.ts

```diff
@@ -0,0 +1,104 @@
+// Type definitions for panel 0.10.0
+// Project: panel
+// Definitions by: Mixpanel (https://mixpanel.com)
+import {VNode} from 'snabbdom';
+import {WebComponent} from 'webcomponent';
+
+export class Component extends WebComponent {
+    /**
+     *  Defines the state of the component, including all the properties required for rendering.
+     */
+    state: any;
+    /**
+     *  Defines standard component configuration.
+     */    
+    config: Component.ComponentConfigOptions;
+    /**
+     *  Template helper functions defined in config object, and exposed to template code as $helpers.
+     *  This getter uses the component's internal config cache.
+     */
+    helpers: any;
+    /**
+     * Called when an instance of the Component is attached to the DOM.
+     */
+     attachedCallback(): void;
+    /**
+     * Called when an attribute is changed, appended, removed, or replaced on the Component.
+     */
+     attributeChangedCallback(attributeName: string, oldValue: any, newValue: any);
+    /**
+     * Called when an instance of the Component is created.
+     */
+     createdCallback(): void;
+    /**
+     * Called when an instance of the Component is detached from the DOM.
+     */
+     detachedCallback(): void;
+    /**
+     * For use inside view templates, to create a child Panel component nested under this
+     * component, which will share its state object and update cycle.
```

Comment:
so are these comments just gonna have to stay in sync manually with the inline JSDoc in the module itself?